models/result.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2021 Greenbone Networks GmbH
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package models
+
+import (
+	"github.com/greenbone/eulabeia/messages"
+	"github.com/greenbone/eulabeia/messages/info"
+)
+
+// Got* structures in this file do not respond to any event. Results and host/scan status messages are published in the way they are generated.
+
+// ResultType is used to identify a result
+type ResultType string
+
+const (
+	HOST_COUNT  ResultType = "HOST_COUNT"  // Result is a host count
+	DEADHOST    ResultType = "DEADHOST"    // Result is a dead host
+	HOST_START  ResultType = "HOST_START"  // Result is a host start
+	HOST_END    ResultType = "HOST_END"    // Result is a host end
+	ERRMSG      ResultType = "ERRMSG"      // Result is a error message
+	LOG         ResultType = "LOG"         // Result is a log message
+	HOST_DETAIL ResultType = "HOST_DETAIL" // Result is a host detail result
+	ALARM       ResultType = "ALARM"       // Result is an alarm result
+)
+
+// Result
+type Result struct {
+	ScanId   string     `json:"scan_id"`  // Scan id
+	Type     ResultType `json:"type"`     // Result type
```

Comment:
`json:"type"` is clashing with `type` from message within the actual json; we need to think about either renaming: type within Message to `message_type` or this `type` to `result_type`.